resolver/resolver.go

```diff
@@ -519,57 +517,55 @@ func (res *Resolver) LaunchHTTP() <-chan error {
 
 // Reports configuration through REST interface
 func (res *Resolver) RestConfig(req *restful.Request, resp *restful.Response) {
-	output, err := json.Marshal(res.config)
-	if err != nil {
+	if err := resp.WriteAsJson(res.config); err != nil {
 		logging.Error.Println(err)
 	}
-
-	io.WriteString(resp, string(output))
 }
 
 // Reports Mesos-DNS version through REST interface
 func (res *Resolver) RestVersion(req *restful.Request, resp *restful.Response) {
-	mapV := map[string]string{"Service": "Mesos-DNS",
+	err := resp.WriteAsJson(map[string]string{
+		"Service": "Mesos-DNS",
 		"Version": res.version,
-		"URL":     "https://github.com/mesosphere/mesos-dns"}
-	output, err := json.Marshal(mapV)
+		"URL":     "https://github.com/mesosphere/mesos-dns",
+	})
 	if err != nil {
 		logging.Error.Println(err)
 	}
-	io.WriteString(resp, string(output))
 }
 
 // Reports Mesos-DNS version through http interface
 func (res *Resolver) RestHost(req *restful.Request, resp *restful.Response) {
-
 	host := req.PathParameter("host")
 	// clean up host name
 	dom := strings.ToLower(cleanWild(host))
 	if dom[len(dom)-1] != '.' {
 		dom += "."
 	}
-
-	mapH := make([]map[string]string, 0)
 	rs := res.records()
 
+	type hostIP struct {
+		Host string `json:"host"`
+		IP   string `json:"ip"`
+	}
+
+	hostIPs := make([]hostIP, 0, len(rs.As[dom]))
 	for _, ip := range rs.As[dom] {
-		t := map[string]string{"host": dom, "ip": ip}
-		mapH = append(mapH, t)
+		hostIPs = append(hostIPs, hostIP{dom, ip})
 	}
-	empty := (len(rs.As[dom]) == 0)
-	if empty {
-		t := map[string]string{"host": "", "ip": ""}
-		mapH = append(mapH, t)
+
+	if len(hostIPs) == 0 {
+		hostIPs = append(hostIPs, hostIP{})
 	}
 
-	output, err := json.Marshal(mapH)
-	if err != nil {
+	if err := resp.WriteAsJson(hostIPs); err != nil {
 		logging.Error.Println(err)
 	}
-	io.WriteString(resp, string(output))
 
-	// stats
-	mesosrq := strings.HasSuffix(dom, res.config.Domain+".")
+	stats(strings.HasSuffix(dom, res.config.Domain+"."), len(rs.As[dom]) == 0)
+}
+
+func stats(mesosrq, empty bool) {
```

Comment:
I took "refactorism" here a little too far perhaps and maintained the previous names, even though they weren't the best. In fact, all I wanted was to remove the code duplication and fix the warning, not properly redesign the code which will be the same thing to do in the long run. Anyways, I'll make it a little better and give more descriptive names to these arguments.